### ISSUE TITLE
Clean up the calls to memcpy

### ISSUE THREAD

**Issue by harpreetbamrah** [eclipse-openj9/openj9#14981]
In furtherance to the technical debt of code, this one is to change the occurences of:
`memcpy( warmMethodHeader + offsetof(OMR::CodeCacheMethodHeader, _metaData), &metaData, sizeof(metaData) );`
to something like:
`reinterpret_cast<OMR::CodeCacheMethodHeader *>(warmMethodHeader)->_metaData = metaData;`

In the on-going MicroJIT PR https://github.com/eclipse-openj9/openj9/pull/9578, this has already been handled, however at existing sources like runtime/compiler/control/CompilationThread.cpp or the like, similar code need to be cleaned up.

**Comment by Brijesh-09** [comment]
I would like to work on this issue

**Comment by akshaywakhare** [comment]
Hi @harpreetbamrah!, If this is still available, I can take it up.

**Comment by harpreetbamrah** [comment]
Thanks @akshaywakhare for the interest, go ahead!

Tagging @dsouzai, just for information.

**Comment by rhyadav** [comment]
Hi @harpreetbamrah  seems like it's open, can i go ahead?

**Comment by harpreetbamrah** [comment]
Sure @rhyadav, if @akshaywakhare is not working on it.

**Comment by rhyadav** [comment]
@akshaywakhare are you working on it? if not i'll open a new issue and start working on it. Thanks @harpreetbamrah

**Comment by adbhonsle** [comment]
@harpreetbamrah Would like to work on this if no one is working on it.
@rhyadav @akshaywakhare

**Comment by harpreetbamrah** [comment]
> @dsouzai @harpreetbamrah Would like to work on this if no one is working on it. @rhyadav @akshaywakhare

sure, go ahead @adbhonsle !

**Comment by adbhonsle** [comment]
@harpreetbamrah The gist is to replace `memcpy` with `reinterpret_cast` right?

**Comment by harpreetbamrah** [comment]
> @harpreetbamrah The gist is to replace `memcpy` with `reinterpret_cast` right?

The gist is to incorporate copy operation using the operator -> directly instead of memcpy function, @dsouzai please correct me if I'm wrong.

**Comment by dsouzai** [comment]
Yeah the point is to not use `memcpy` when we have an embedded struct, but rather just using the `=` operator so that we're not doing pointer math on things that don't need pointer math.

**Comment by adbhonsle** [comment]
Just to be sure this need not be replaced, right?
`memcpy(current, J9WRONGNAME, sizeof(J9WRONGNAME) - 1);`

**Comment by dsouzai** [comment]
Yeah data that is not described by a data structure will still need the use of `memcpy`; this issue is more to track places where the structure is defined but we still chose to use pointers to copy data.

**Comment by adbhonsle** [comment]
Hi @dsouzai 
What is U_32 & U_8 and how are they related ? 
If possible could you tell where definition is?

**Comment by dsouzai** [comment]
`U_32` and `U_8` are just primitive types that are consistent across all the platforms; `U_32` is an unsigned 32-bit integer, and `U_8` is an unsigned 8-bit integer. I believe the definitions are in https://github.com/eclipse/omr/blob/master/include_core/omrcomp.h
